src/main/java/com/o19s/es/ltr/ranker/dectree/NaiveAdditiveDecisionTree.java

```diff
@@ -155,4 +166,63 @@ public long ramBytesUsed() {
             return BASE_RAM_USED;
         }
     }
+
+    public static class BuildState {
+        Node[] trees;
+        float[] weights;
+        int modelSize;
+
+        public BuildState(Node[] trees, float[] weights, int modelSize) {
+            this.trees = trees;
+            this.weights = weights;
+            this.modelSize = modelSize;
+        }
+
+        public NaiveAdditiveDecisionTree build() {
+            Node[] res = new Node[trees.length];
+            for (int i = 0; i < trees.length; i++) {
+                res[i] = fixMinLeafValue(trees[i]);
+            }
+            return new NaiveAdditiveDecisionTree(res, this.weights, this.modelSize);
+        }
+
+        /**
+         * Adjust the minimum leaf value within trees to MIN_LEAF_VALUE. Various
+         * learning algos will build trees that return negative values, but that
+         * can cause issues when combining ltr scores with non-ltr scores, or
+         * even just when zeroing out non-ltr scores as the negative values
+         * may sort last.
+         *
+         * @param tree A tree to fix the minimum value of
+         * @return A new tree with fixed minimum value.
+         */
+        private Node fixMinLeafValue(Node tree) {
+            float minLeaf = minLeaf(tree);
+            if (minLeaf >= MIN_LEAF_VALUE) {
+                return tree;
+            } else {
+                return addToLeafs(tree, -minLeaf + MIN_LEAF_VALUE);
```

Comment:
if meanLeaf is [0,MIN_LEAF_VALUE] you'll have scores lower than MIN_LEAF_VALUE, you still ensure that they're all positive but not lower than MIN_LEAF_VALUE